remote-sr/src/main/java/de/unistuttgart/iaas/amyassist/amy/remotesr/RemoteSR.java

```diff
@@ -212,20 +253,37 @@ void setClient(SSEClient client) {
 	 *
 	 * @return true if the request was successful
 	 */
-	public boolean requestSR() {
-		return this.client != null && client.sendEvent(START_SR_EVENT);
+	boolean tryToRecognize() {
+		if (this.client == null) {
+			this.logger.error("Could not start recognition, because I have no client.");
+			return false;
+		} else if (!this.client.isConnected()) {
+			this.logger.error("Could not start recognition, because the client is not connected.");
+			return false;
+		}
+		this.currentlyRecognizing = true;
+		return this.client.sendEvent(START_SR_EVENT);
 	}
 
+	/**
+	 * Processes a result from the speech recognition.
+	 * 
+	 * @param res
+	 *            The result.
+	 */
 	void processResult(SRResult res) {
-		if (res != null && res.getText() != null && this.listener != null) {
-			this.listener.remoteSRDidRecognizeSpeech(res.getText());
+		if (res != null && res.getText() != null && this.listener != null && !res.getText().isEmpty()) {
```

Comment:
An empty string should still be passed to the listener. Is signalises that the user did not say anything within a reasonable listening time (maybe 4 seconds).